File: src/main/java/org/lineageos/sleepaudio/ui/SleepAudioTileService.java
package org.lineageos.sleepaudio.ui;

import android.content.SharedPreferences;
import android.service.quicksettings.Tile;
import android.service.quicksettings.TileService;
import androidx.preference.PreferenceManager;
import org.lineageos.sleepaudio.utils.Constants;

public class SleepAudioTileService extends TileService {
    @Override
    public void onStartListening() {
        super.onStartListening();
        updateTile();
    }

    @Override
    public void onClick() {
        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
        boolean enabled = prefs.getBoolean(Constants.KEY_ENABLE, false);
        prefs.edit().putBoolean(Constants.KEY_ENABLE, !enabled).apply();
        updateTile();
    }

    private void updateTile() {
        Tile tile = getQsTile();
        if (tile == null) return;
        
        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
        boolean enabled = prefs.getBoolean(Constants.KEY_ENABLE, false);
        
        tile.setState(enabled ? Tile.STATE_ACTIVE : Tile.STATE_INACTIVE);
        tile.updateTile();
    }
}

File: app/src/main/kotlin/org/lineageos/sleepaudio/ui/SleepAudioTileService.kt
package org.lineageos.sleepaudio.ui

import android.service.quicksettings.Tile
import android.service.quicksettings.TileService
import androidx.preference.PreferenceManager
import org.lineageos.sleepaudio.utils.Constants

class SleepAudioTileService : TileService() {

    override fun onClick() {
        val prefs = PreferenceManager.getDefaultSharedPreferences(this)
        val state = prefs.getBoolean(Constants.KEY_ENABLE, false)
        prefs.edit().putBoolean(Constants.KEY_ENABLE, !state).apply()
        updateTile()
    }

    override fun onStartListening() {
        super.onStartListening()
        updateTile()
    }

    private fun updateTile() {
        val tile = qsTile ?: return
        val prefs = PreferenceManager.getDefaultSharedPreferences(this)
        val state = prefs.getBoolean(Constants.KEY_ENABLE, false)
        
        tile.state = if (state) Tile.STATE_ACTIVE else Tile.STATE_INACTIVE
        tile.label = "SleepAudio"
        tile.subtitle = if (state) "Active" else "Off"
        tile.updateTile()
    }
}
